### PR TITLE
PWGHF: fix the PID in the Lc->pKpi selector

### PR DESCRIPTION
**Pull request by DelloStritto**

### CODE REVIEW

**Comment by vkucera** [comment]
As discussed, please adjust the decay strings and use named PDG values where possible.

File: Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx [hunk 1]
```diff
@@ -328,27 +332,41 @@ struct HFLcCandidateSelector {
...
-      if (proton == 0 || kaonMinus == 0 || pionPlus == 0) {
-        pidLc = 0; //exclude Lc
-      }
-      if (proton == 1 && kaonMinus == 1 && pionPlus == 1) {
-        pidLc = 1; //accept Lc
+      if (!d_FilterPID) {
+        // PID non applied
+        pidLcpKpi = 1;
+        pidLcpiKp = 1;
+      } else {
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx]: It must be sad to call `selectionPID` five times and then realise that `!d_FilterPID`. :'-(

**Review by vkucera — CHANGES_REQUESTED** [review]

File: Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx (hunk 1 above)
Comment (DelloStritto) [reply Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx]: Good Point! I can move them inside the else. Thanks!

**Comment by vkucera** [comment]
Also please fix `/// \file HFD0CandidateSelector.cxx`.

**Comment by DelloStritto** [comment]
> As discussed, please adjust the decay strings and use named PDG values where possible.

Done!

File: Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx [hunk 2]
```diff
@@ -281,7 +282,7 @@ struct HFLcCandidateSelector {
   {
     int statusLcpKpi, statusLcpiKp; // final selection flag : 0-rejected  1-accepted
     bool topolLcpKpi, topolLcpiKp;
-    int pidLc, proton, kaonMinus, pionPlus;
+    int pidLcpKpi, pidLcpiKp, proton1, proton2, kaonMinus, pionPlus1, pionPlus2;
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx]: Most of these variables don't need to be declared here and initialised a bit later. Instead they can be declared and initialised directly with the results of the related functions.

**Comment by vkucera** [comment]
Can you please also adjust the decay strings to `LcToPKPi` in the code, file names, function names, and columns and tables in `hf_selcandidate_lc`?

**Comment by DelloStritto** [comment]
> Can you please also adjust the decay strings to `LcToPKPi` in the code, file names, function names, and columns and tables in `hf_selcandidate_lc`?

I would prefer to do it in a separate pull request. These change will include a lot of files that are already involved in other pull request so I will generate a lot of conflicts.  
If it's fine with you I would start merging the PID pull request and then I can open a "clean up" pull request with the name changes.

**Comment by vkucera** [comment]
> > Can you please also adjust the decay strings to `LcToPKPi` in the code, file names, function names, and columns and tables in `hf_selcandidate_lc`?
> 
> I would prefer to do it in a separate pull request. These change will include a lot of files that are already involved in other pull request so I will generate a lot of conflicts.
> If it's fine with you I would start merging the PID pull request and then I can open a "clean up" pull request with the name changes.

OK, as you prefer. Let's keep it for the big renaming commit then.

**Review by vkucera — APPROVED** [review]
@ginnocen Looks good to me.

**Comment by ginnocen** [comment]
@vkucera please resolve the conversations if closed.

**Review by ginnocen — APPROVED** [review]

**Pull request merged**